### ISSUE TITLE
StreamTransport does not verify SSL certificates

### ISSUE THREAD

**Issue by cebe** [yiisoft/yii2-httpclient#222]
### What steps will reproduce the problem?

https://github.com/yiisoft/yii2-httpclient/blob/3fc9b07b413ddddc4f74822b58df9b18f639f6f0/src/StreamTransport.php#L41-L43

### What's expected?

This value should be configurable via a property and the default should be `true` instead of `false`.

### Additional info

| Q                       | A
| ----------------------- | ---
| Yii version             | not relevant
| Yii HTTP Client version | all
| PHP version             | not relevant
| Operating system        | all

**Comment by schmunk42** [comment]
Do you mean a standalone property? Because isn't it configurable via `options`, see [here](https://github.com/yiisoft/yii2-httpclient/blob/3fc9b07b413ddddc4f74822b58df9b18f639f6f0/src/StreamTransport.php#L53)?

I agree about the security implications, but it also might break several exisiting apps.

**Comment by cebe** [comment]
> Because isn't it configurable via `options`, see [here](https://github.com/yiisoft/yii2-httpclient/blob/3fc9b07b413ddddc4f74822b58df9b18f639f6f0/src/StreamTransport.php#L53)?

good point, did not notice it was configurable like that.

It is "only" a bad default then.

**Comment by samdark** [comment]
Yes, default should be adjusted. No need to add a dedicated property though.
